Automattic/woocommerce-payments
# Make adding fee details to order notes async

Fixes #4439 

#### Changes proposed in this Pull Request

- Makes adding fee details to order notes async.
- Removes the get timeline call from the checkout flow for performance benefits, ref: pdpOw2-Sb-p2#comment-1140
- We are adding a new order note instead of updating the existing one since:
  - it might be inconsistent behavior for the end-user to see a note with different information after refreshing in case they visit before and after the async job runs - the queue can take some time to pick the scheduled tasks even if we schedule it immediately.
  - From what I could see, we also don't have any precedence of updating notes in the codebase, but please CMIIAW here.








#### Testing instructions


- Follow the normal checkout flow and look at the order details screen in the WooCommerce tab.
- Ensure there is a separate order note added to the order. Go to Tools > Scheduled actions > search for this job with `wcpay_add_fee_breakdown_to_order_notes` and select `Run` for running the action.


Before:

<img width="280" alt="Screenshot 2022-07-12 at 5 20 36 PM" src="https://user-images.githubusercontent.com/68396823/178483512-63cc22fe-5e4c-4ac0-8f02-af9c42d3ff28.png">

After:

<img width="279" alt="Screenshot 2022-07-12 at 5 08 10 PM" src="https://user-images.githubusercontent.com/68396823/178483546-d567a4f8-ab4d-4041-8d14-15570516d73f.png">

-------------------


**Post merge**



- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : [Testing instructions](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions-for-WC-Payments-4.5.0#make-adding-fee-details-to-order-notes-async)
